geopyspark/geotrellis/key_conversion.py

```diff
@@ -0,0 +1,157 @@
+        self.__layout = self.__jvm.geotrellis.spark.tiling.LayoutDefinition(ex, tilelayout)
+
+    def key_to_extent(self, key, *args):
+        """Returns the Extent corresponding to a given key.
+
+        Args:
+            key (:class:`~geopyspark.geotrellis.SpatialKey` or :class:`~geopyspark.geotrellis.SpaceTimeKey` or int): The
+                key to find the extent for.  If of type int, then this parameter is the column of the key, and the call
+                must provide a single additional int value in the args parameter to serve as the row of the key.
+
+        Returns:
+            :class:`~geopyspark.geotrellis.Extent`
+        """
+        if isinstance(key, (gps.SpatialKey, gps.SpaceTimeKey)):
+            skey = self.__jvm.geotrellis.spark.SpatialKey(key.col, key.row)
+        elif isinstance(key, tuple):
+            skey = self.__jvm.geotrellis.spark.SpatialKey(key[0], key[1])
+        elif isinstance(key, int) and len(args) == 1 and isinstance(args[0], int):
+            skey = self.__jvm.geotrellis.spark.SpatialKey(key, args[0])
+        else:
+            raise ValueError("Please supply either gps.SpatialKey, gps.SpaceTimeKey, (int, int), or two ints")
+        ex = self.__layout.mapTransform().apply(skey)
+        return gps.Extent(ex.xmin(), ex.ymin(), ex.xmax(), ex.ymax())
+
+    def extent_to_keys(self, extent):
+        """Returns the keys in the layout intersecting/covered by a given extent.
+
+        Args:
+            extent (:class:`~geopyspark.geotrellis.Extent`): The extent to find the matching keys for.
+
+        Returns:
+            [:class:`~geopyspark.geotrellis.SpatialKey`]
+        """
+        ex = self.__jvm.geotrellis.vector.Extent(float(extent.xmin), float(extent.ymin), float(extent.xmax), float(extent.ymax))
+        gridbnd = self.__layout.mapTransform().apply(ex)
+        cmin = gridbnd.colMin()
+        cmax = gridbnd.colMax()
+        rmin = gridbnd.rowMin()
+        rmax = gridbnd.rowMax()
+        return (gps.SpatialKey(c, r) for c in range(cmin, cmax + 1) for r in range(rmin, rmax + 1))
+
+    def geometry_to_keys(self, geom):
+        """Returns the keys corresponding to grid cells that intersect/are covered by a given Shapely geometry.
+
+        Args:
+            geom (:class:`~shapely.geometry.Geometry`): The geometry to find the matching keys for.
+
+        Returns:
+            [:class:`~geopyspark.geotrellis.SpatialKey`]
+        """
+        from shapely.wkb import dumps
+        jts_geom = self.__jvm.geopyspark.geotrellis.util.GeometryUtil.wkbToScalaGeometry(dumps(geom))
+        scala_key_set = self.__layout.mapTransform().keysForGeometry(jts_geom)
+        key_set = self.__jvm.scala.collection.JavaConverters.setAsJavaSetConverter(scala_key_set).asJava()
+        return [gps.SpatialKey(key.col(), key.row()) for key in key_set]
+
+    """The WebMercator (EPSG=3857) world extent"""
```

Comment:
I think that these constants should be moved out of the class and into the module. Since there's not really a reason for each class instance to have their own constants.